packages/server/src/scripts/import_arpa_reporter_dump.js

```diff
@@ -0,0 +1,469 @@
+        .value();
+
+    return { inserted, idLookup };
+}
+
+async function importAgencies(
+    dbContents,
+    idLookupByTable,
+    insertedRowsByTable
+) {
+    // First, create all agencies, defaulting their parent pointer to point to themselves.
+    const agenciesToCreate = dbContents.agencies.map((agency) =>
+        rekeyForeignKeys(
+            {
+                tenant_id: agency.tenant_id,
+                name: agency.name,
+                code: agency.code,
+                abbreviation: agency.code,
+                parent: knex.ref("id"),
+                main_agency_id: knex.ref("id"),
+            },
+            idLookupByTable,
+            ["main_agency_id", "parent"]
+        )
+    );
+    let inserted = await knex("agencies")
+        .insert(agenciesToCreate)
+        .returning("*");
+    const idLookup = _.chain(inserted)
+        .map((insertedRow, idx) => [
+            dbContents.agencies[idx].id,
+            insertedRow.id,
+        ])
+        .fromPairs()
+        .value();
+
+    // Then update main_agency_id on tenant rows
+    for (const tenant of insertedRowsByTable.tenants) {
+        const mainAgencyId = idLookup[tenant.__unmappedFutureMainAgencyId];
+        tenant.main_agency_id = mainAgencyId;
+        await knex("tenants")
+            .where("id", tenant.id)
+            .update({ main_agency_id: mainAgencyId });
+    }
+
+    // Then update any non-main agencies to be parented by their tenant's main agency (since
+    // standalone ARPA Reporter did not have a concept of parent agencies)
+    const agencyIds = _.map(inserted, "id");
+    await knex.raw(
+        `
+        UPDATE agencies
+        SET
+            parent = tenants.main_agency_id,
+            main_agency_id = tenants.main_agency_id,
+        FROM tenants
+        WHERE
+            agencies.tenant_id = tenants.id
+            AND agencies.id in :agencyIds
+        `,
+        { agencyIds }
+    );
+    inserted = await knex("agencies").select("*").whereIn("id", agencyIds);
+
+    return { inserted, idLookup };
+}
+
+async function importUsers(dbContents, idLookupByTable, insertedRowsByTable) {
+    const roles = await knex('roles').select('*');
+    const adminRole = roles.find(r => r.name == 'admin').id;
+    const staffRole = roles.find(r => r.name == 'staff').id;
+
+    const mainAgencyByTenant = _.chain(insertedRowsByTable.tenants)
+        .keyBy("id")
+        .mapValues("main_agency_id")
+        .value();
+    const usersToCreate = dbContents.users.map((user) =>
+        rekeyForeignKeys(
+            {
+                email: user.email,
+                name: user.name,
+                // Note: ARPA Reporter had a "reporter" role. For our purposes,
+                // anything non-admin becomes "staff"
+                role_id: user.role === 'admin' ? adminRole : staffRole,
+                tenant_id: user.tenant_id,
+                // Copied users belong to the main agency of their tenant
+                agency_id:
```

Comment:
In the ARPA tables admin users have an empty agency id, and reporter type users have specific ids set. Are there likely to be consequences for changing that for either/both of these user types?

Reply:
I guess I should take a second pass over ARPA code to see if there's any behavior that's gated on `user.agency_id` -- my impression before was that all permissions are already at tenant scope in ARPA.

Reply:
As far as I can tell from grepping for `.agency_id`, `user.agency_id` is only used for display -- do you know of anything more consequential I've missed @ajhyndman?

Reply:
Huh, it looks like you're right.

I had thought that we restricted upload visibility to uploads belonging to your agency.  I just tested though, and that is definitely not the case.